diff -crB a/fs/dcache.c b/fs/dcache.c
*** a/fs/dcache.c	2025-12-19 15:24:39.324297154 +0000
--- b/fs/dcache.c	2025-12-19 15:26:16.078180774 +0000
***************
*** 2219,2224 ****
--- 2219,2229 ----
  			continue;
  		if (dentry_cmp(dentry, str, hashlen_len(hashlen)) != 0)
  			continue;
+ #ifdef CONFIG_KSU_SUSFS_SUS_PATH
+ 			if (dentry->d_inode && unlikely(dentry->d_inode->i_state & INODE_STATE_SUS_PATH) && likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC)) {
+ 				continue;
+ 			}
+ #endif
  		*seqp = seq;
  		return dentry;
  	}
diff -urB a/fs/notify/fdinfo.c b/fs/notify/fdinfo.c
--- a/fs/notify/fdinfo.c	2025-12-19 17:01:21.879382671 +0000
+++ b/fs/notify/fdinfo.c	2025-12-19 17:03:01.442778621 +0000
@@ -108,6 +108,37 @@
 		 * used only internally to the kernel.
 		 */
 		u32 mask = mark->mask & IN_ALL_EVENTS;
+#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+		if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) &&
+				unlikely(inode->i_state & INODE_STATE_SUS_KSTAT)) {
+			struct path path;
+			char *pathname = kmalloc(PAGE_SIZE, GFP_KERNEL);
+			char *dpath;
+			if (!pathname) {
+				goto out_seq_printf;
+			}
+			dpath = d_path(&file->f_path, pathname, PAGE_SIZE);
+			if (!dpath) {
+				goto out_free_pathname;
+			}
+			if (kern_path(dpath, 0, &path)) {
+				goto out_free_pathname;
+			}
+			seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:0 ",
+			   inode_mark->wd, path.dentry->d_inode->i_ino, path.dentry->d_inode->i_sb->s_dev,
+			   inotify_mark_user_mask(mark));
+			show_mark_fhandle(m, path.dentry->d_inode);
+			seq_putc(m, '\n');
+			iput(inode);
+			path_put(&path);
+			kfree(pathname);
+			return;
+out_free_pathname:
+			kfree(pathname);
+		}
+out_seq_printf:
+#endif
 		seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:%x ",
 			   inode_mark->wd, inode->i_ino, inode->i_sb->s_dev,
 			   mask, mark->ignored_mask);
diff -crB a/fs/readdir.c b/fs/readdir.c
*** a/fs/readdir.c	2025-12-19 15:24:39.324297154 +0000
--- b/fs/readdir.c	2025-12-19 15:29:27.161814818 +0000
***************
*** 414,419 ****
--- 414,424 ----
  
  	if (buf->result)
  		return -EINVAL;
+ #ifdef CONFIG_KSU_SUSFS_SUS_PATH
+ 	if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) && susfs_sus_ino_for_filldir64(ino)) {
+ 		return 0;
+ 	}
+ #endif
  	d_ino = ino;
  	if (sizeof(d_ino) < sizeof(ino) && d_ino != ino) {
  		buf->result = -EOVERFLOW;
