diff -crB a/fs/notify/fdinfo.c b/fs/notify/fdinfo.c
*** a/fs/notify/fdinfo.c	2025-12-10 09:22:46.330356108 +0000
--- b/fs/notify/fdinfo.c	2025-12-10 09:27:50.943965217 +0000
***************
*** 83,88 ****
--- 83,119 ----
  	inode_mark = container_of(mark, struct inotify_inode_mark, fsn_mark);
  	inode = igrab(mark->inode);
  	if (inode) {
+ #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
+ 		u32 mask = mark->mask & IN_ALL_EVENTS;
+ 		if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) &&
+ 				unlikely(inode->i_state & INODE_STATE_SUS_KSTAT)) {
+ 			struct path path;
+ 			char *pathname = kmalloc(PAGE_SIZE, GFP_KERNEL);
+ 			char *dpath;
+ 			if (!pathname) {
+ 				goto out_seq_printf;
+ 			}
+ 			dpath = d_path(&file->f_path, pathname, PAGE_SIZE);
+ 			if (!dpath) {
+ 				goto out_free_pathname;
+ 			}
+ 			if (kern_path(dpath, 0, &path)) {
+ 				goto out_free_pathname;
+ 			}
+ 			seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:%x ",
+ 			   inode_mark->wd, path.dentry->d_inode->i_ino, path.dentry->d_inode->i_sb->s_dev,
+ 			   mask, mark->ignored_mask);
+ 			show_mark_fhandle(m, path.dentry->d_inode);
+ 			seq_putc(m, '\n');
+ 			iput(inode);
+ 			path_put(&path);
+ 			kfree(pathname);
+ 			return;
+ out_free_pathname:
+ 			kfree(pathname);
+ 		}
+ out_seq_printf:
+ #endif
  		seq_printf(m, "inotify wd:%x ino:%lx sdev:%x mask:%x ignored_mask:0 ",
  			   inode_mark->wd, inode->i_ino, inode->i_sb->s_dev,
  			   inotify_mark_user_mask(mark));
diff -crB a/fs/proc/cmdline.c b/fs/proc/cmdline.c
*** a/fs/proc/cmdline.c	2025-12-10 09:22:46.330356108 +0000
--- b/fs/proc/cmdline.c	2025-12-10 09:29:56.954249688 +0000
***************
*** 2,7 ****
--- 2,12 ----
  #include <linux/init.h>
  #include <linux/proc_fs.h>
  #include <linux/seq_file.h>
+ 
+ #ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+ extern int susfs_spoof_cmdline_or_bootconfig(struct seq_file *m);
+ #endif
+ 
  #ifdef CONFIG_INITRAMFS_IGNORE_SKIP_FLAG
  #include <asm/setup.h>
  #endif
***************
*** 28,33 ****
--- 33,45 ----
  
  static int cmdline_proc_show(struct seq_file *m, void *v)
  {
+ #ifdef CONFIG_KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+ 	if (!susfs_spoof_cmdline_or_bootconfig(m)) {
+ 		seq_putc(m, '\n');
+ 		return 0;
+ 	}
+ #endif
+ 
  #ifdef CONFIG_INITRAMFS_IGNORE_SKIP_FLAG
  	seq_printf(m, "%s\n", proc_command_line);
  #else
diff -crB a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
*** a/fs/proc/task_mmu.c	2025-12-10 09:22:46.330356108 +0000
--- b/fs/proc/task_mmu.c	2025-12-10 09:30:35.914952942 +0000
***************
*** 17,22 ****
--- 17,25 ----
  #include <linux/shmem_fs.h>
  #include <linux/mm_inline.h>
  #include <linux/ctype.h>
+ #ifdef CONFIG_KSU_SUSFS_SUS_KSTAT
+ #include <linux/susfs_def.h>
+ #endif
  
  #include <asm/elf.h>
  #include <asm/uaccess.h>
