--- a/drivers/media/platform/msm/sde/rotator/sde_rotator_dev.c	2025-12-11 19:56:34.225099517 +0000
+++ b/drivers/media/platform/msm/sde/rotator/sde_rotator_dev.c	2025-12-11 19:57:13.381490814 +0000
@@ -1304,11 +1304,7 @@
 
 	req = &rot_mdata->pm_qos_rot_cpu_req;
 	req->type = PM_QOS_REQ_AFFINE_CORES;
-	cpumask_empty(&req->cpus_affine);
-	for_each_possible_cpu(cpu) {
-		if ((1 << cpu) & cpu_mask)
-			cpumask_set_cpu(cpu, &req->cpus_affine);
-	}
+	atomic_set(&req->cpus_affine, cpu_mask);
 	pm_qos_add_request(req, PM_QOS_CPU_DMA_LATENCY,
 		PM_QOS_DEFAULT_VALUE);
 
