--- a/techpack/display/msm/msm_cooling_device.c	2025-12-08 22:38:10.410522788 +0700
+++ b/techpack/display/msm/msm_cooling_device.c	2025-12-08 22:55:09.214583596 +0700
@@ -1,19 +1,17 @@
 // SPDX-License-Identifier: GPL-2.0-only
 /*
- * Copyright (c) 2020-2021 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2020, The Linux Foundation. All rights reserved.
  */
 #include <linux/err.h>
 #include <linux/slab.h>
 #include "msm_cooling_device.h"
 
-#define BRIGHTNESS_CDEV_MAX 255
-
 static int sde_cdev_get_max_brightness(struct thermal_cooling_device *cdev,
 					unsigned long *state)
 {
 	struct sde_cdev *disp_cdev = (struct sde_cdev *)cdev->devdata;
 
-	*state = disp_cdev->bd->props.max_brightness / disp_cdev->cdev_sf;
+	*state = disp_cdev->bd->props.max_brightness;
 
 	return 0;
 }
@@ -23,8 +21,7 @@
 {
 	struct sde_cdev *disp_cdev = (struct sde_cdev *)cdev->devdata;
 
-	*state = ((disp_cdev->bd->props.max_brightness -
-			disp_cdev->thermal_state) / disp_cdev->cdev_sf);
+	*state = disp_cdev->bd->props.max_brightness - disp_cdev->thermal_state;
 
 	return 0;
 }
@@ -35,11 +32,10 @@
 	struct sde_cdev *disp_cdev = (struct sde_cdev *)cdev->devdata;
 	unsigned long brightness_lvl = 0;
 
-	if (state > disp_cdev->bd->props.max_brightness / disp_cdev->cdev_sf)
+	if (state > disp_cdev->bd->props.max_brightness)
 		return -EINVAL;
 
-	brightness_lvl = disp_cdev->bd->props.max_brightness -
-				(state * disp_cdev->cdev_sf);
+	brightness_lvl = disp_cdev->bd->props.max_brightness - state;
 	if (brightness_lvl == disp_cdev->thermal_state)
 		return 0;
 	disp_cdev->thermal_state = brightness_lvl;
@@ -72,12 +68,6 @@
 	disp_cdev->thermal_state = 0;
 	disp_cdev->bd = bd;
 
-	if (bd->props.max_brightness > BRIGHTNESS_CDEV_MAX)
-		disp_cdev->cdev_sf = (bd->props.max_brightness /
-						BRIGHTNESS_CDEV_MAX);
-	else
-		disp_cdev->cdev_sf = 1;
-
 	disp_cdev->cdev = thermal_of_cooling_device_register(dev->of_node,
 				(char *)dev_name(&bd->dev), disp_cdev,
 				&sde_cdev_ops);
--- a/techpack/display/msm/msm_cooling_device.h	2025-12-08 22:38:10.410522788 +0700
+++ b/techpack/display/msm/msm_cooling_device.h	2025-12-08 22:55:37.950583585 +0700
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: GPL-2.0-only */
 /*
- * Copyright (c) 2020-2021 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2020, The Linux Foundation. All rights reserved.
  */
 
 #ifndef __SDE_THERMAL_CORE_H__
@@ -16,7 +16,6 @@
 	struct thermal_cooling_device *cdev;
 	struct backlight_device *bd;
 	unsigned long thermal_state;
-	unsigned int cdev_sf;
 };
 
 #ifdef CONFIG_THERMAL_OF
